src/views/Upload.vue

```diff
@@ -219,61 +233,45 @@ export default {
       this.upload = null
       this.errors = []
 
-      try {
-        const resp = await fetch(`/api/uploads/${this.uploadId}`)
-        const result = (await resp.json()) || { error: (await resp.body) }
-
-        if (resp.ok) {
-          this.upload = result.upload
-        } else {
-          this.alert = {
-            text: `loadUpload Error (${resp.status}): ${result.error}`,
-            level: 'err'
-          }
-        }
-      } catch (e) {
-        this.alert = {
-          text: `loadUpload Unknown Error: ${e.message}`,
+      const result = await getJson(`/api/uploads/${this.uploadId}`)
+      if (result.error) {
+        this.$store.commit('addAlert', {
+          text: `loadUpload Error (${result.status}): ${result.error}`,
           level: 'err'
-        }
+        })
+      } else {
+        this.upload = result.upload
       }
 
       // each time we refresh the upload, also refresh the series
       this.loadSeries()
     },
     loadSeries: async function () {
-      try {
-        const resp = await fetch(`/api/uploads/${this.uploadId}/series`)
-        const result = (await resp.json()) || { error: (await resp.body) }
+      this.series = []
+      this.seriesValid = null
 
-        if (resp.ok) {
-          this.series = result.series
-          this.seriesValid = result.currently_valid
-        } else {
-          this.alert = {
-            text: `loadUpload API Error (${resp.status}): ${result.error}`,
-            level: 'err'
-          }
-        }
-      } catch (e) {
-        this.alert = {
-          text: `loadUpload Unknown Error: ${e.message}`,
+      const result = await getJson(`/api/uploads/${this.uploadId}/series`)
+      if (result.error) {
+        this.$store.commit('addAlert', {
+          text: `loadSeries Error (${result.status}): ${result.error}`,
```

Comment:
Another possible future enhancement:

Is there detail in this request that isn't present in the `allUploads` store cache?  I think it'd be nice if this could be implemented as a `getter` on the store instead of triggering another server round trip.

Reply:
i have to double-check the DB code, we might not be consistent in how we load associations -- specifically, the user email address, and the agency code.